File: spojPractice/spoj2148.java
import java.io.*;
import java.math.*;
import java.util.*;
class spoj2148
{
	public static void main(String args[])throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		int t=Integer.parseInt(br.readLine());
		for(int q=1;q<=t;q++)
		{
			System.out.println();
			BigInteger n=new BigInteger(br.readLine());
			//BigInteger a[]=new BigInteger[n];
			BigInteger sum=BigInteger.valueOf(0);
			for(BigInteger i=BigInteger.valueOf(0);i.compareTo(n)<0;i=i.add(BigInteger.valueOf(1)))
			{
				BigInteger number=new BigInteger(br.readLine());
				sum=sum.add(number);
				//sum=sum.add(a[i]);
			}
			if(sum.mod(n).compareTo(BigInteger.valueOf(0))==0)
				System.out.println("YES");
			else
				System.out.println("NO");
		}
	}
}

File: spojPractice/spoj2148.cpp
#include <iostream>
using namespace std;
int main()
{
	long long int t;
	cin>>t;
	while(t--)
	{ 
		cout<<endl;
		long long int n;
		cin>>n;
		long long int *m;
		m=new long long int[n];
		long long int sum=0;
		for(long long int i=0;i<n;i++)
		{
			long long int p;
			cin>>p;
			sum+=p;
			if(sum>=n)
				sum=sum%n;
		}
		if(sum%n==0)
			cout<<"YES\n";
		else
			cout<<"NO\n";
	}
	return 0;
}